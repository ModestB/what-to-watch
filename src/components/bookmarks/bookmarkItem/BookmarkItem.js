import React from 'react';

import classes from './BookmarkItem.module.scss';
import CloseIcon from '../../../icons/js/Close';

const BookmarkItem  = React.memo(function BookmarkItem (props) {
  function clickHandler () {
    props.displayBookmarksHandler();
    props.findShowById(props.id, props.mediaType);
  }

  return (
    <div className='d-flex justify-content-between align-items-center pb-2'>
      <div 
        className={`${classes.title} d-flex text-left`}
        onClick={clickHandler}
      >
        {`${props.title} (${props.date})`}
      </div>
      
      <CloseIcon 
        className={`${classes.closeIcon}`}
        onClick={() => props.removeBookmark(props.id)}
        width="20px" 
        height="20px"
      />
    </div>
  )
});

export default BookmarkItem;