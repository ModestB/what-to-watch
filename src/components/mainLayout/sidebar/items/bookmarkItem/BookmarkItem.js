import React, { useEffect, useState, useRef } from "react";
import { connect } from "react-redux";

import classes from "./BookmarkItem.module.scss";

import HeartIcon from "../../../../../icons/js/Heart";
import SidebarItem from "../SidebarItem";

export const BookmarksItem = (props) => {
  const isInitialMount = useRef(true);
  let [heartKey, setHeartKey] = useState(0);
  useEffect(() => {
    if (isInitialMount.current) {
      isInitialMount.current = false;
    } else {
      setHeartKey(heartKey + 1);
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [props.bookmarks]);

  return (
    <SidebarItem title="Bookmarks" route="bookmarks">
      <HeartIcon
        key={heartKey > 1 ? heartKey : "0"}
        fill="#ffffff"
        width="30px"
        height="30px"
        className={classes.iconHeart}
      />
    </SidebarItem>
  );
};

const mapStateProps = (state) => {
  return {
    bookmarks: state.bookmarks,
  };
};

export default connect(mapStateProps)(BookmarksItem);
