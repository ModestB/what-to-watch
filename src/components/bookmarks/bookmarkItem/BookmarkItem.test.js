import React from "react";

import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import { BookmarkItem } from "./BookmarkItem";

configure({ adapter: new Adapter() });

describe("<BookmarkItem />", () => {
  let wrapper;
  let bookmark = {
    id: 420809,
    title: "Maleficent: Mistress of Evil",
    date: "2019",
    mediaType: "movie",
  };

  beforeEach(() => {
    wrapper = shallow(<BookmarkItem {...bookmark} />);
  });

  it("should render initial layout", () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("title should match props.title", () => {
    expect(wrapper.find(".title").text()).toEqual(
      `${bookmark.title} (${bookmark.date})`
    );
  });
});
