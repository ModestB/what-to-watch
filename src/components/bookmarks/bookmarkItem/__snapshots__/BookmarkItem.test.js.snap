// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookmarkItem /> should render initial layout 1`] = `
"<div className=\\"container\\">
  <div className=\\"title\\" onClick={[Function: clickHandler]}>
    Maleficent: Mistress of Evil (2019)
  </div>
  <SvgClose className=\\"icon-close icon-close--danger\\" onClick={[Function: onClick]} width=\\"20px\\" height=\\"20px\\" />
</div>"
`;
